array_api_strict/_typing.py

```diff
@@ -8,82 +8,66 @@
 
 from __future__ import annotations
 
-__all__ = [
-    "Array",
-    "Device",
-    "Dtype",
-    "SupportsDLPack",
-    "SupportsBufferProtocol",
-    "PyCapsule",
-]
-
 import sys
+from typing import Any, Protocol, TypedDict, TypeVar
 
-from typing import (
-    Any,
-    TypedDict,
-    TypeVar,
-    Protocol,
-)
-
-from ._array_object import Array, _device
-from ._dtypes import _DType
-from ._info import __array_namespace_info__
+from ._dtypes import DType
 
 _T_co = TypeVar("_T_co", covariant=True)
 
+
 class NestedSequence(Protocol[_T_co]):
     def __getitem__(self, key: int, /) -> _T_co | NestedSequence[_T_co]: ...
     def __len__(self, /) -> int: ...
 
 
-Device = _device
-
-Dtype = _DType
-
-Info = __array_namespace_info__
-
 if sys.version_info >= (3, 12):
     from collections.abc import Buffer as SupportsBufferProtocol
 else:
     SupportsBufferProtocol = Any
 
 PyCapsule = Any
 
+
 class SupportsDLPack(Protocol):
     def __dlpack__(self, /, *, stream: None = ...) -> PyCapsule: ...
 
+
 Capabilities = TypedDict(
-    "Capabilities", {"boolean indexing": bool, "data-dependent shapes": bool,
-                     "max dimensions": int}
+    "Capabilities",
+    {
+        "boolean indexing": bool,
+        "data-dependent shapes": bool,
+        "max dimensions": int,
+    },
 )
 
 DefaultDataTypes = TypedDict(
     "DefaultDataTypes",
     {
-        "real floating": Dtype,
-        "complex floating": Dtype,
-        "integral": Dtype,
-        "indexing": Dtype,
+        "real floating": DType,
+        "complex floating": DType,
+        "integral": DType,
+        "indexing": DType,
     },
 )
 
 DataTypes = TypedDict(
     "DataTypes",
     {
-        "bool": Dtype,
-        "float32": Dtype,
-        "float64": Dtype,
-        "complex64": Dtype,
-        "complex128": Dtype,
-        "int8": Dtype,
-        "int16": Dtype,
-        "int32": Dtype,
-        "int64": Dtype,
-        "uint8": Dtype,
-        "uint16": Dtype,
-        "uint32": Dtype,
-        "uint64": Dtype,
+        "bool": DType,
+        "float32": DType,
+        "float64": DType,
+        "complex64": DType,
+        "complex128": DType,
+        "int8": DType,
+        "int16": DType,
+        "int32": DType,
+        "int64": DType,
+        "uint8": DType,
+        "uint16": DType,
+        "uint32": DType,
+        "uint64": DType,
```

Comment:
if there are no spaces, the general recommendation is to use the class-based syntax instead

```py
class DataTypes(TypedDict, total=false):
	bool: DType
	float32: DType
	float64: DType
	complex64: DType
	complex128: DType
	int8: DType
	int16: DType
	int32: DType
	int64: DType
	uint8: DType
	uint16: DType
	uint32: DType
	uint64: DType
```

I think there's something in `ruff` that would complain about it, as well 🤷🏻